File: src/main/java/me/kuku/yuq/dao/QQLoginDao.java
package me.kuku.yuq.dao;

import com.icecreamqaq.yudb.YuDao;
import com.icecreamqaq.yudb.jpa.annotation.Dao;
import com.icecreamqaq.yudb.jpa.annotation.Execute;
import com.icecreamqaq.yudb.jpa.annotation.Select;
import me.kuku.yuq.entity.QQLoginEntity;

import java.util.List;

@Dao
public interface QQLoginDao extends YuDao<QQLoginEntity, Integer> {
    QQLoginEntity findByQQ(Long qq);
    @Select("from QQLoginEntity")
    List<QQLoginEntity> findAll();
    @Select("from QQLoginEntity where status = true")
    List<QQLoginEntity> findByActivity();
    @Execute("delete from QQLoginEntity where qq = ?0")
    void delByQQ(Long qq);
}

File: src/main/kotlin/me/kuku/yuq/dao/QQLoginDao.kt
package me.kuku.yuq.dao

import com.icecreamqaq.yudb.YuDao
import com.icecreamqaq.yudb.jpa.annotation.Dao
import com.icecreamqaq.yudb.jpa.annotation.Execute
import com.icecreamqaq.yudb.jpa.annotation.Select
import me.kuku.yuq.entity.QQLoginEntity

@Dao
interface QQLoginDao: YuDao<QQLoginEntity, Int>{
    fun findByQQ(qq: Long): QQLoginEntity?
    @Select("from QQLoginEntity")
    fun findAll(): List<QQLoginEntity>
    @Select("from QQLoginEntity where status = true")
    fun findByActivity(): List<QQLoginEntity>
    @Execute("delete from QQLoginEntity where qq = ?0")
    fun delByQQ(qq: Long)
}